test/blackbox-tests/test-cases/pkg/portable-lockdirs/portable-lockdirs-basic.t

```diff
@@ -0,0 +1,161 @@
+   ((arch x86_64)
+    (os win32)
+    (os-distribution cygwin)
+    (os-family windows))
+   ((arch arm64)
+    (os win32)
+    (os-distribution cygwin)
+    (os-family windows)))
+
+  $ cat dune.lock/foo.pkg
+  (version 0.0.1)
+  
+  (build
+   (choice
+    ((((arch x86_64)
+       (os linux)
+       (os-distribution ubuntu)
+       (os-family debian))
+      ((arch x86_64)
+       (os linux)
+       (os-distribution alpine)
+       (os-family alpine))
+      ((arch x86_64)
+       (os linux)))
+     ((action
+       (progn
+        (run mkdir -p %{share} %{lib}/%{pkg-self:name})
+        (run touch %{lib}/%{pkg-self:name}/META)
+        (run sh -c "echo Linux > %{share}/kernel")
+        (run sh -c "echo x86_64 > %{share}/machine")))))
+    ((((arch arm64)
+       (os linux)
+       (os-distribution ubuntu)
+       (os-family debian))
+      ((arch arm64)
+       (os linux)
+       (os-distribution alpine)
+       (os-family alpine))
+      ((arch arm64)
+       (os linux)))
+     ((action
+       (progn
+        (run mkdir -p %{share} %{lib}/%{pkg-self:name})
+        (run touch %{lib}/%{pkg-self:name}/META)
+        (run sh -c "echo Linux > %{share}/kernel")
+        (run sh -c "echo arm64 > %{share}/machine")))))
+    ((((arch x86_64) (os macos) (os-distribution homebrew) (os-family homebrew)))
+     ((action
+       (progn
+        (run mkdir -p %{share} %{lib}/%{pkg-self:name})
+        (run touch %{lib}/%{pkg-self:name}/META)
+        (run sh -c "echo Darwin > %{share}/kernel")
+        (run sh -c "echo x86_64 > %{share}/machine")))))
+    ((((arch arm64) (os macos) (os-distribution homebrew) (os-family homebrew)))
+     ((action
+       (progn
+        (run mkdir -p %{share} %{lib}/%{pkg-self:name})
+        (run touch %{lib}/%{pkg-self:name}/META)
+        (run sh -c "echo Darwin > %{share}/kernel")
+        (run sh -c "echo arm64 > %{share}/machine")))))
+    ((((arch x86_64) (os win32) (os-distribution cygwin) (os-family windows)))
+     ((action
+       (progn
+        (run mkdir -p %{share} %{lib}/%{pkg-self:name})
+        (run touch %{lib}/%{pkg-self:name}/META)
+        (run sh -c "echo x86_64 > %{share}/machine")))))
+    ((((arch arm64) (os win32) (os-distribution cygwin) (os-family windows)))
+     ((action
+       (progn
+        (run mkdir -p %{share} %{lib}/%{pkg-self:name})
+        (run touch %{lib}/%{pkg-self:name}/META)
+        (run sh -c "echo arm64 > %{share}/machine")))))))
+
+  $ DUNE_CONFIG__OS=linux DUNE_CONFIG__ARCH=arm64 DUNE_CONFIG__OS_FAMILY=debian DUNE_CONFIG__OS_DISTRIBUTION=ubuntu DUNE_CONFIG__OS_VERSION=24.11 dune build
+  $ cat _build/_private/default/.pkg/foo/target/share/kernel
+  Linux
+  $ cat _build/_private/default/.pkg/foo/target/share/machine
+  arm64
+
+  $ DUNE_CONFIG__OS=macos DUNE_CONFIG__ARCH=x86_64 DUNE_CONFIG__OS_FAMILY=homebrew DUNE_CONFIG__OS_DISTRIBUTION=homebrew DUNE_CONFIG__OS_VERSION=15.3.1 dune build
+  $ cat _build/_private/default/.pkg/foo/target/share/kernel
```

Comment:
Could you avoid using the magic paths in the tests? We have some shell helpers to do this. You could introduce new ones if the ones we have here aren't enough.